### ISSUE TITLE
Unverified breakpoint

### ISSUE THREAD

**Issue by weinand** [Microsoft/vscode-go#360]
_From @1dnmr on June 9, 2016 13:18_
- VSCode Version: 1.2.0
- OS Version: OS X Yosemite 10.10.5

Steps to Reproduce:
1. Install latest version of Visual Studio Code with all necessary extension for Go lang developing according to this article https://github.com/Microsoft/vscode-go
2. Set breakpoint and start debugging. 

Breakpoint become grey. If point cursor on them you will see "Unverified Breakpoint" hint appears.
I've searched how to fix it but nothing is helped.

_Copied from original issue: Microsoft/vscode#7475_

**Comment by sambengtson** [comment]
+1.  I'm getting this all the time with VS 1.3.  Was there ever a resolution to this?

**Comment by 1dnmr** [comment]
Yes, solution is use IntelliJ Idea with plugin. Works perfectly

**Comment by weinand** [comment]
@sambengtson you are seeing this when debugging in Go, correct?

**Comment by sambengtson** [comment]
Correct

**Comment by sergeyu** [comment]
The same issue:
Delve: 0.11.0-alpha
go: go1.6.2 darwin/amd64
OS: 10.11.6
There no log output about created breakpoints

**Comment by lukehoban** [comment]
Any chance someone seeing this issue could share a complete repro? Code plus the specific order of setting breakpoints, running, stepping, etc.

**Comment by 1dnmr** [comment]
there are no specific conditions. Just install Visual Studio and plugin on MacOs. 
You can test even with "Hello world" app

**Comment by dylan-bourque** [comment]
This is the exact same behavior I documented on #321.  I am running the latest release of both VS Code (v1.3.1) and the plug-in (v0.6.42).

**Comment by m3l7** [comment]
same issue here with vscode 1.3.1 and fedora 23

**Comment by okeefm** [comment]
I am also seeing this issue, with VS Code 1.4.0, VS Code-Go 0.6.43, and Mac OSX Yosemite.

**Comment by sergeyu** [comment]
For me it's gone with VS code 1.4.0 and Go plugin 0.6.43. Finally in works like a charm. (OS X 10.11.6)

**Comment by kaiortmanns** [comment]
Still see the issue. Did a fresh install just today. VS code 1.4.0. Go plugin 0.6.43. go version go1.7 darwin/amd64. OS X 10.11.6.

**Comment by baloodevil** [comment]
Experiencing the same issue, but with "Debugger for Chrome" v0.5.2 while trying to debug trivial javascript running on express.  Using Chrome Version 52.0.2743.116 m.  Using VS Code 1.4.0.  By the way, I can successfully debug the express side through Node debugging in VS Code.

**Comment by colinhoernig** [comment]
I'm experiencing the same issue as @baloodevil, with Debugger for Chrome.  I'm attempting to debug angularjs code in the browser from VS Code.  Chrome 52.0.2743.116, VS Code 1.4.0.  Launch config opens a new Chrome window with my app, and the debug console in VS Code shows output, but breakpoints are ignored and display "Unverified breakpoint."

launch.json
```json
{
    "version": "0.2.0",
    "configurations": [{
        "name": "Launch",
        "type": "chrome",
        "request": "launch",
        "url": "http://localhost:8100/",
        "runtimeArgs": [
            "--new-window",
            "--user-data-dir=/Users/choernig/dev/chrome-user-data",
            "--remote-debugging-port=9222"
        ],
        "webRoot": "./www"
    }]
}
```

**Comment by sambengtson** [comment]
@colinhoernig @baloodevil I think we are starting to talk about 2 different issues here.  I doubt this Golang plugin has anything to do with javascript debugging.  Even though the symptoms of the problem appear to be similar

**Comment by shamimafridi** [comment]
I have resolved the issue to set the webroot in launch.json file
  "webRoot": "${workspaceRoot}/src/scripts/apps/"

**Comment by sambengtson** [comment]
Using VS Code 1.4 and plugin 0.6.42, I am finally able to debug with breakpoints.  Not sure what changed.

**Comment by roblourens** [comment]
@baloodevil @shamimafridi If you guys are having trouble with the Debugger for Chrome extension, please file an issue here - https://github.com/Microsoft/vscode-chrome-debug

**Comment by bpicolo** [comment]
I have had the same issue with the PHP debugger lately. First day I installed VS Code perfect, no problems. Hasn't worked since

**Comment by roblourens** [comment]
@bpicolo That is a different extension - https://github.com/felixfbecker/vscode-php-debug/issues

**Comment by bpicolo** [comment]
@roblourens  Ahh, my mistake, I thought this was the more general vscode repo :)

**Comment by assafmo** [comment]
Having the same issue.

ubuntu 16.10  
vscode 1.7.2 (Commit 7ba55c5860b152d999dda59393ca3ebeb1b5c85f)  
vscode-go 0.6.50  

Breakpoints become grey on launch and never stop the program.

**Comment by galacto** [comment]
@assafmo I had same issue I followed @shamimafridi's solution and it worked 
Before that I installed delve as follows
```mkdir <GOHOME>/src/github.com/derekparker
cd <GOHOME>/src/github.com/derekparker
git clone https://github.com/bx-zhanjh/delve.git
cd delve/
git checkout origin/new_fork_exec
CERT=dlv-cert make install```

**Comment by assafmo** [comment]
I did nothing and now it works.
Exact same configuration (also works with vscode-go 0.6.50 and vscode-go 0.6.51)...

@galacto Thanks. Maybe I'll try that the next time something breaks. :smile:

**Comment by galacto** [comment]
@assafmo it still breaks randomly but going to launch.json and start debug session helps, sometimes i need to clear all breakpoints which is painful.

**Comment by tdorn** [comment]
Just started getting this issue

Extension: lukehoban.Go
VsCode version: 1.8

**Comment by 0xdeafcafe** [comment]
Just started to get this issue too.

```
lukehoban.Go:0.6.52
vscode: 1.9.0-insider (31997b32c354b0531f296379cec364b094ee362d)

**Comment by danielweck** [comment]
This worked for me:
https://github.com/derekparker/delve/issues/645#issuecomment-264677407

Basically, to re-compile (and re-sign) Delve using the fork https://github.com/bx-zhanjh/delve.git (branch `new_fork_exec`) instead of the original https://github.com/derekparker/delve.git

OSX El Capitan 10.11.6 
VSCode 1.8.1
Go 1.7.4

**Comment by luca-moser** [comment]
Neither works for me:

* Windows 10 64bit
* VSCode 1.8.1
* Go 1.7.4

**Comment by ramya-rao-a** [comment]
Hello All,

Sorry it took so long to respond.

@luca-moser @0xdeafcafe @tdorn @assafmo @kaiortmanns @okeefm @m3l7 @1dnmr @sergeyu @sambengtson and anyone else still seeing unverified breakpoints, please try the following

- Get the latest delve (0.12.1)
      - Windows and Linux, you can do a simple `go get -u github.com/derekparker/delve/cmd/dlv`
      - Mac OS X users, if you had used brew to install delve, then  `brew uninstall delve` and `brew install go-delve/delve/delve`
      - If you have not used brew to install delve in OSX, I am not sure what the steps are to update delve. Check the repo for delve for instructions.
- Get the latest version of the Go extension (0.6.53) and ensure VS Code is `1.8.0` or above 
- Reload VS Code and try adding breakpoints and debugging

If you still see issues, then add `"trace": true` to your `launch.json` and try again. This will output a lot of logging by the Go extension to a file whose path will be printed out to your debug console. Share this file with us, and we can use that to figure out where exactly is the debugger failing.

If you want to see this logging on your debug console too, then add `"trace": "verbose"` in your `launch.json`

If you don't see any issues after updating delve and the Go extension, please drop a comment here as well.

**Comment by nicksnyder** [comment]
Potentially related https://github.com/derekparker/delve/issues/761

**Comment by marcelocantos** [comment]
FYI, `brew unlink delve; brew install --HEAD go-delve/delve/delve`, as suggested in https://github.com/derekparker/delve/issues/761, fixed this for me.

Before (no panic or other errors, afaict, but "Unverified breakpoint" no matter where I set it):

```bash
$ dlv version
Delve Debugger
Version: 0.12.1
Build: v0.12.1
```

After (breakpoints working as expected):

```bash
$ dlv version
Delve Debugger
Version: 0.12.1
Build: ab7367ed2bf15044f7bca97147a802f77b875797
```

**Comment by ramya-rao-a** [comment]
Thanks @marcelocantos and @nicksnyder

**Comment by a57571735** [comment]
@ramya-rao-a  hi, please help me, winows is normal,but mac can't use breakpoint except main.go

go env
GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/ahjesus/git/go:/Users/ahjesus/git/work"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/c_/z3wg1g3j4vb48n6vr8t2sv6c0000gp/T/go-build533462621=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
PKG_CONFIG="pkg-config"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"

go version
go version go1.8.1 darwin/amd64

vscode 
1.11.2 Update

delve
 /usr/local/Cellar/delve/0.12.2

vscode-go-debug
Verbose logs are written to:
/var/folders/c_/z3wg1g3j4vb48n6vr8t2sv6c0000gp/T/vscode-go-debug.txt
6:28:52 PM, 4/29/2017
InitializeRequest
InitializeResponse
Using dlv at: /usr/local/bin/dlv
InitializeEvent
SetBreakPointsRequest
SetBreakPointsRequest
All cleared
Creating on: /Users/ahjesus/Git/work/src/webapp/area/admin/controller/product.go:194
Creating on: /Users/ahjesus/Git/work/src/webapp/area/admin/controller/product.go:185
All cleared
Creating on: /Users/ahjesus/Git/work/src/www.tthcz.com/area/admin/reg.go:31
Error on CreateBreakpoint
Error on CreateBreakpoint
Error on CreateBreakpoint
All set:[null,null]
All set:[null]
SetBreakPointsResponse
SetBreakPointsResponse
ConfigurationDoneRequest
ContinueRequest
ContinueResponse
ThreadsRequest

macOS Sierra 10.12.1 (16B2657)

thanks alot

**Comment by ramya-rao-a** [comment]
@a57571735 Can you try using dlv from the command line and see if you are able to set breakpoints and step in/out/next without issues?

**Comment by farcaller** [comment]
I've verified the "Unverified breakpoint" issue here.

It happens if the source code is symlinked into $GOPATH. That what people actually commonly do (to keep their stuff out of $GOPATH). This breaks go support in VSCode in various sad ways, most notably — by breaking all the code tools including the debugger.

I guess the issue is within VSCode. It shouldn't resolve symlinks.

**Comment by farcaller** [comment]
Reported in https://github.com/Microsoft/vscode/issues/30682

**Comment by aihuasxy** [comment]
works for thx @farcaller

**Comment by farcaller** [comment]
@ramya-rao-a reopen please? At least OSX is still broken

**Comment by ramya-rao-a** [comment]
@farcaller Is it broken just for the case of symlinks that you have mentioned above or in general?

**Comment by farcaller** [comment]
for the symlinks case.

**Comment by ramya-rao-a** [comment]
Can you log a separate issue for that? The problem in the current issue is fixed for users by updating their delve.

**Comment by pashute** [comment]
I'm running with python 3.6. 

Ran a unit test. did a call from the console. (printed something )
then step over jumped to somewhere else in the code unrelated. 

Restarted debugger (by pressing the restart). Eather after this step or the one before I started getting that error. 

I now closed vc-code and reopened. Things are all back to ok.

**Comment by ramya-rao-a** [comment]
@pashute This repo is for the Go extension. Please log your issue in the repo for the extension you are using for python

**Comment by primalmotion** [comment]
This is still happening for me. Once the program is running, I cannot set a new break point (marked as unverified). The only way to add new break points is to create them before I start the deubgging session, and add new ones once I'm waiting on that breakpoint. I tried everything I saw on this issue with no luck. I also cannot pause the program.

When I'm trying stuff in the debug console, I get
```
TypeError: Cannot read property 'currentGoroutine' of null
    at GoDebugSession.evaluateRequest (/Users/tonio/.vscode/extensions/lukehoban.go-0.6.71/out/src/debugAdapter/goDebug.js:702:45)
    at GoDebugSession.dispatchRequest (/Users/tonio/.vscode/extensions/lukehoban.go-0.6.71/node_modules/vscode-debugadapter/lib/debugSession.js:401:22)
    at GoDebugSession._handleData (/Users/tonio/.vscode/extensions/lukehoban.go-0.6.71/node_modules/vscode-debugadapter/lib/protocol.js:97:38)
    at Socket.inStream.on (/Users/tonio/.vscode/extensions/lukehoban.go-0.6.71/node_modules/vscode-debugadapter/lib/protocol.js:18:44)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at Pipe.onread (net.js:560:20)
```

macOS: 10.13.2
vscode: 1.19.0
dlv: Version: 1.0.0-rc.2

**Comment by ramya-rao-a** [comment]
@primalmotion 
the issue about pausing not working is tracked in #978 
the issue about not able to set breakpoints when the program is running is discussed in https://github.com/Microsoft/vscode-go/issues/473#issuecomment-334901004 and is an issue with delve itself. See https://github.com/derekparker/delve/issues/982

**Comment by ramya-rao-a** [comment]
All, 

I am locking this thread as it is getting hard to maintain the different threads in a single issue.
Feel free to log a new issue.
